File: src/main/java/no/nav/tjenestepensjon/simulering/exceptions/StillingsprosentCallableException.java
package no.nav.tjenestepensjon.simulering.exceptions;

import java.util.concurrent.ExecutionException;

import no.nav.tjenestepensjon.simulering.domain.TPOrdning;

public class StillingsprosentCallableException extends ExecutionException {

    private final TPOrdning tpOrdning;

    public StillingsprosentCallableException(String msg, Throwable cause, TPOrdning tpOrdning) {
        super(msg, cause);
        this.tpOrdning = tpOrdning;
    }

    public TPOrdning getTpOrdning() {
        return tpOrdning;
    }

    @Override
    public String toString() {
        return "StillingsprosentCallableException{" +
                "message'" + getLocalizedMessage() + '\'' +
                "tpOrdning='" + tpOrdning.toString() + '\'' +
                '}';
    }
}

File: src/main/kotlin/no/nav/tjenestepensjon/simulering/exceptions/StillingsprosentCallableException.kt
package no.nav.tjenestepensjon.simulering.exceptions

import no.nav.tjenestepensjon.simulering.model.domain.TPOrdning

class StillingsprosentCallableException(private val msg: String, cause: Throwable, val tpOrdning: TPOrdning) : Throwable(msg, cause) {

    override fun toString() =
            "StillingsprosentCallableException{message'$msg'tpOrdning='$tpOrdning'}"
}